net-p2p/retroshare/retroshare-0.6.5.ebuild

```diff
@@ -0,0 +1,141 @@
+# Copyright 1999-2019 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGIT_REPO_URI="https://github.com/RetroShare/RetroShare.git"
+EGIT_COMMIT="v${PV}"
```

Comment:
Upstream seems to make proper releases as seen in https://github.com/RetroShare/RetroShare/releases, so why do you use `git-r3` when you should make `SRC_URI` point at a source tarball instead?

Reply:
Because the tarball are automatically generated by GitHub some parts of the build system (as example `jsonapi`) are broken because it depends on .git being present and working

Reply:
then pull a snapshot from github instead of using `git-r3`.

Reply:
The problem with the snapshot is that is miss the .git so git commands executed as part of the retroshare build-system would fail. Git is used for non trivial tasks as part of the build system. It could be someway avoided, but it would mean rewrite part of the retroshare build-system inside the ebuild, with the risk of introducing new bugs etc... I don't like the situation either, but after investigating the things using `git-r3` seems the most reasonable solution.

Reply:
Which features are broken if you use the upstream tarball?

Reply:
As example JSON API and retroshare-service are broken, more in general anything that is a submodule in retroshare repository or that depends on a submodule is broken. If we really want to avoid `git-r3` usage, either the ebuild get the submodules at correct version in the ebuild using plain git commands, or I propose to retroshare develoment team to prepare and distribute a source tarball including them. The problem is that `git archive` doesn't offer such a  functionality so a custom script that prepare a complete source tarball need to be written.

Reply:
If they don't want to be a packaging nightmare, they have to include their submodules in the release tarball. An alternative for you would be to provide the necessary submodule snapshots in SRC_URI.

Reply:
Should be fixed now, thanks!